batch.md

```diff
@@ -48,52 +62,76 @@ DataSource<EventType> dataSet = env.createInput(inputFormat, TypeInformation.of(
+(can u rephrase??)
 ### Parallelism
 `FlinkPravegaInputFormat` supports parallelization. Use the `setParallelism` method of `DataSet` to configure the number of parallel instances to execute.  The parallel instances consume the stream in a coordinated manner, each consuming one or more stream segments.
 
-_Note: Coordination is achieved with the use of a Pravega reader group, which is based on a [State Synchronizer](http://pravega.io/docs/pravega-concepts/#state-synchronizers).  The synchronizer creates a backing stream that may be manually deleted after the job finishes._
+```java
+final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+         env.setParallelism(3);
 
+```
 
+**Note**: Coordination is achieved with the use of a Pravega reader group, which is based on a [State Synchronizer](http://pravega.io/docs/pravega-concepts/#state-synchronizers). The synchronizer creates a backing stream that may be manually deleted after the job finishes.
+
+### Builder API Usage
+A builder API is provided to construct an instance of `FlinkPravegaInputFormat`.  See the table below for a summary of builder properties.  Note that the builder accepts an instance of `PravegaConfig` for common configuration properties.  See the [configurations](configurations) page for more information.
+
+|Method                |Description|
+|----------------------|-----------------------------------------------------------------------|
+|`withPravegaConfig`|The Pravega client configuration, which includes connection info, security info, and a default scope.|
+|`forStream`|The stream to be read from, with optional start and/or end position.  May be called repeatedly to read numerous streams in parallel.|
+|`withDeserializationSchema`|The deserialization schema which describes how to turn byte messages into events.|
 
-## Data Sink (OutputFormat)
-A Pravega stream may be used as a data sink within a Flink batch program using an instance of `io.pravega.connectors.flink.FlinkPravegaOutputFormat`.  The `FlinkPravegaOutputFormat` can be supplied as a sink to the [`DataSet`](https://ci.apache.org/projects/flink/flink-docs-master/api/java/org/apache/flink/api/java/DataSet.html#output-org.apache.flink.api.common.io.OutputFormat-) (the basic abstraction of the Flink Batch API).
 
-Here's an example:
 
+## FlinkPravegaOutputFormat
+A Pravega stream may be used as a data sink within a Flink batch program using an instance of 
+```
+io.pravega.connectors.flink.FlinkPravegaOutputFormat.
+
+```
+The `FlinkPravegaOutputFormat` can be supplied as a sink to the [`DataSet`](https://ci.apache.org/projects/flink/flink-docs-master/api/java/org/apache/flink/api/java/DataSet.html#output-org.apache.flink.api.common.io.OutputFormat-) (the basic abstraction of the Flink Batch API).
+
+### Example
+
+Pravega configuration can be defined using:

```

Comment:
I actually prefer to keep a single block of code instead of breaking it up like below
```
// Define the Pravega configuration
PravegaConfig config = PravegaConfig.fromParams(params);

// Define the event serializer
SerializationSchema<EventType> serializer = ...

// Define the event router for selecting the routing key
PravegaEventRouter<EventType> router = ...

// Define the input format based on a Pravega stream
FlinkPravegaOutputFormat<EventType> outputFormat = FlinkPravegaOutputFormat.<EventType>builder()
    .forStream(...)
    .withPravegaConfig(config)
    .withSerializationSchema(serializer)
    .withEventRouter(router)
    .build();

ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
Collection<EventType> inputData = Arrays.asList(...);
env.fromCollection(inputData)
   .output(outputFormat);
env.execute("...");
```

Reply:
Accepted suggestion